gensim/models/tfidfmodel.py

```diff
@@ -387,8 +392,24 @@ def __getitem__(self, bow, eps=1e-12):
 
         # and finally, normalize the vector either to unit length, or use a
         # user-defined normalization function
-        vector = self.normalize(vector)
+        norm_vector = self.normalize(vector)
 
-        # make sure there are no explicit zeroes in the vector (must be sparse)
-        vector = [(termid, weight) for termid, weight in vector if abs(weight) > eps]
-        return vector
+        # Need to check if self.pivot_norm variable is in the local scope or not to
+        # mantain backward compatibility.
+        if hasattr(self, 'pivot_norm') and self.pivot_norm is True:
```

Comment:
This is really strange check, better to do all needed "backward-compatible" stuff in `load` function (to avoid confusion in main code).